api-tests/src/test/java/org/openmrs/module/reporting/cohort/definition/evaluator/DrugOrderCohortDefinitionEvaluatorTest.java

```diff
@@ -0,0 +1,185 @@
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs before " + cohort.getMemberIds() + " " + cohortDefinition);
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertEquals(1, cohort.size());
+
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsTakingAnyDrugWithinADateRange() throws Exception {
+    cohortDefinition.setActivatedOnOrAfter(DateUtil.getDateTime(2008, 8, 1));
+    cohortDefinition.setActivatedOnOrBefore(DateUtil.getDateTime(2008, 8, 8));
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs in date range " + cohort.getMemberIds() + " " + cohortDefinition);
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertTrue(cohort.contains(7));
+    Assert.assertTrue(cohort.contains(8));
+    Assert.assertTrue(cohort.contains(21));
+    Assert.assertTrue(cohort.contains(22));
+    Assert.assertEquals(3, cohort.size());
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsTakingSpecifiedDrugBeforeDate() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setActivatedOnOrBefore(DateUtil.getDateTime(2008, 8, 2));
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Particular Drug in Date Range" + cohort.getMemberIds() + " " + cohortDefinition);
+    Assert.assertTrue(cohort.contains(2));
+
+  }
+
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsActiveOnDrugsWithinLastSpecifiedMonths() throws Exception {
+    cohortDefinition.setOnlyCurrentlyActive(true);
+    cohortDefinition.setActiveWithinLastMonths(1);
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println(new Date());
+    System.out.println(
+        "Specified months" + cohort.getCommaSeparatedPatientIds() + " " + cohortDefinition);
+
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsActiveOnDrugsWithinLastSpecifiedDays() throws Exception {
+    cohortDefinition.setOnlyCurrentlyActive(true);
+    cohortDefinition.setActiveWithinLastDays(1);
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println(new Date());
+    System.out.println("Specified days" + cohort.getCommaSeparatedPatientIds() + " " + cohortDefinition);
+
+  }
+
+  @Ignore
+  @Test
+  public void evaluate_shouldReturnAllInSpecifiedCareSetting() throws Exception {
+    cohortDefinition.setOnlyCurrentlyActive(true);
+    // cohortDefinition.setCareSetting(careSetting);
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Care Setting" + cohort.getMemberIds());
```

Comment:
Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout